src/stan/mcmc/hmc/base_hmc.hpp

```diff
@@ -134,6 +134,10 @@ class base_hmc : public base_mcmc {
 
   typename Hamiltonian<Model, BaseRNG>::PointType& z() { return z_; }
 
+  const typename Hamiltonian<Model, BaseRNG>::PointType& z() const noexcept {
```

Comment:
1. Needs documentation.  This is very simple here.  There may not be doc elsewhere, but we're requiring it for new additions and appreciate it getting backfilled where it's missing.  

2. Needs a test.  This can be really simple, but we just want to make sure we have tests in place to avoid future regressions.

Reply:
Done.